modules/account-lib/src/coin/dot/transaction.ts

```diff
@@ -1,20 +1,202 @@
-import { BaseTransaction } from '../baseCoin';
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { BaseTransaction, TransactionType } from '../baseCoin';
 import { BaseKey } from '../baseCoin/iface';
-import { NotImplementedError } from '../baseCoin/errors';
+import { InvalidTransactionError, SigningError } from '../baseCoin/errors';
+import { construct, decode } from '@substrate/txwrapper-polkadot';
+import { UnsignedTransaction } from '@substrate/txwrapper-core';
+import { TypeRegistry } from '@substrate/txwrapper-core/lib/types';
+import { decodeAddress } from '@polkadot/keyring';
+import { KeyPair } from './keyPair';
+import { TxData, DecodedTx, StakeArgs, StakeArgsPayeeRaw, AddProxyArgs, UnstakeArgs } from './iface';
+import utils from './utils';
 
 export class Transaction extends BaseTransaction {
+  protected _dotTransaction: UnsignedTransaction;
+  private _signedTransaction?: string;
+  private _registry: TypeRegistry;
+  private _chainName: string;
+  private _sender: string;
+
+  constructor(coinConfig: Readonly<CoinConfig>) {
+    super(coinConfig);
+  }
+
```

Comment:
@argjv 
Do we need to add here also 
- numberOfRequiredSignatures
- maxNumberOfSignatures
As the standard mention?

Reply:
i think we're using singlesig, specifying these two would be unnecessary i guess